src/transformStateWithClones.js

```diff
@@ -8,6 +8,36 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const clonedState = [];
+  const currentStateData = { ...state };
+  const addDataToArray = () => {
+    clonedState.push({ ...currentStateData });
+  };
+
+  for (const action in actions) {
+    switch (actions[action].type) {
+      case 'addProperties':
+        Object.assign(currentStateData, { ...actions[action].extraData });
+        addDataToArray();
+        break;
+      case 'removeProperties':
+        for (const keyToRemove in actions[action].keysToRemove) {
+          delete currentStateData[actions[action].keysToRemove[keyToRemove]];
+        }
```

Comment:
this loop is to complicated, and you can make it simpler

Reply:
already Fixed in next commit